plox/plox/interpreter.py

```diff
@@ -0,0 +1,117 @@
+from plox.expressions import Expr, Literal, Grouping, Unary, Binary
+from plox.token_types import TokenType, Token
+from plox.errors import LoxErrors, LoxRuntimeError
+from functools import singledispatchmethod
+
+
+class Interpreter:
+    def interpret(self, expr: Expr):
+        try:
+            value = self._evaluate(expr)
+            print(self._stringify(value))
+        except Exception as e:
```

Comment:
I think this only needs to catch `LoxRuntimeError`s. `LoxErrors.runtime_error` can't handle other kinds of errors well.